src/shared/components/test/Buttons-test.js

```diff
@@ -0,0 +1,343 @@
+import { mount } from 'enzyme';
+
+import { IconButton, LabeledButton, LinkButton } from '../Buttons.js';
+import { $imports } from '../Buttons.js';
+
+//import { checkAccessibility } from '../../../test-util/accessibility';
+import mockImportedComponents from '../../../test-util/mock-imported-components';
+
+describe('IconButton', () => {
+  let fakeOnClick;
+
+  function createComponent(props = {}) {
+    return mount(
+      <IconButton
+        icon="fakeIcon"
+        title="My Action"
+        onClick={fakeOnClick}
+        {...props}
+      />
+    );
+  }
+
+  beforeEach(() => {
+    fakeOnClick = sinon.stub();
+    $imports.$mock(mockImportedComponents());
+  });
+
+  afterEach(() => {
+    $imports.$restore();
+  });
+
+  it('renders the indicated icon', () => {
+    const wrapper = createComponent();
+
+    const icon = wrapper.find('SvgIcon');
+    assert.equal(icon.prop('name'), 'fakeIcon');
+  });
+
+  it('invokes callback on click', () => {
+    const onClick = sinon.stub();
+    const wrapper = createComponent({ onClick });
+
+    wrapper.find('button').simulate('click');
+    assert.calledOnce(onClick);
+  });
+
+  it('uses a default className', () => {
+    const wrapper = createComponent();
+
+    assert.isTrue(wrapper.find('button').hasClass('IconButton'));
+  });
+
+  it('renders a primary variant', () => {
+    const wrapper = createComponent({ variant: 'primary' });
+
+    assert.isTrue(wrapper.find('button').hasClass('IconButton--primary'));
+  });
+
+  it('overrides className when provided', () => {
+    const wrapper = createComponent({
+      className: 'CustomClassName',
+      variant: 'primary',
+    });
+
+    assert.isTrue(wrapper.find('button').hasClass('CustomClassName'));
+    assert.isTrue(wrapper.find('button').hasClass('CustomClassName--primary'));
+    assert.isFalse(wrapper.find('button').hasClass('IconButton'));
+  });
+
+  [
+    {
+      propName: 'expanded',
+      propValue: true,
+      attributeName: 'aria-expanded',
+      attributeValue: 'true',
+    },
+    {
+      propName: 'pressed',
+      propValue: true,
+      attributeName: 'aria-pressed',
+      attributeValue: 'true',
+    },
+    {
+      propName: 'title',
+      propValue: 'Click here',
+      attributeName: 'aria-label',
+      attributeValue: 'Click here',
+    },
+    {
+      propName: 'disabled',
+      propValue: true,
+      attributeName: 'disabled',
+      attributeValue: '',
+    },
+  ].forEach(testCase => {
+    it('sets attributes on the button element', () => {
```

Comment:
There is a bunch of duplication between the different components here. I can suggest two ways around this:

- Move the base button component into a separate module and tests those details there. In this module you'd just test that the props are forwarded correctly.
- Create a function which defines the test cases and is parametrized by the component to render

```js
function addCommonTests(component) {
  // Use `describe`, `it` etc. here to define test cases
}

describe('IconButton', () => {
  ...
  addCommonTests(IconButton);
});

describe('LinkButton', () => {
   ...
   addCommonTests(LinkButton);
});
```

Reply:
I've refactored the tests to de-dupe.